src/commons/interceptors.js

```diff
@@ -28,6 +34,14 @@ function buildAuthHeader(accessToken) {
  * @returns {object}
  */
 export function requestInterceptor(request) {
+  request.cancelToken = new CancelToken(function executor(token) {
+    cancel = token;
+  });
+
+  if (!window.navigator.onLine) {
+    cancel(NETWORK_CONNECTION_ERROR);
```

Comment:
Let's not cancel the request when the user is offline.
Let the request pass through and we can handle the error once we get an invalid response.

Reply:
So can we check for the connection status in the response interceptor? And then reject the promise with custom error for network error?

Reply:
@Krishuthapa 
Nope, not needed. It will fail automatically when you're offline and trigger the `catch` block.

Everytime we make an API call, we have to wrap it in a `try/catch` block, and use the generic error handler. So if the user is offline, it will automatically fail with a `Network Error` message.
```js
try {
 let { data } = await employee.fetch();
 ...
} catch (e) {
  handleError(e) // This block will always run when the user is offline.
}
```